packages/engine/src/utils.rs

```diff
@@ -1,23 +1,104 @@
-use std::{env::VarError, time::Duration};
+use std::{env::VarError, fmt::Display, time::Duration};
 
-pub fn init_logger() {
-    use tracing_subscriber::{fmt::time, prelude::*, EnvFilter};
+use log::{Event, Subscriber};
+use tracing_subscriber::{
+    filter::{Directive, LevelFilter},
+    fmt::{
+        self,
+        format::{Format, Writer},
+        time::FormatTime,
+        FmtContext, FormatEvent, FormatFields,
+    },
+    prelude::*,
+    registry::LookupSpan,
+    EnvFilter,
+};
 
-    if cfg!(debug_assertions) && std::env::var("RUST_LOG").is_err() {
-        std::env::set_var(
-            "RUST_LOG",
-            "hash_cloud=debug,hash_engine=debug,cli=debug,server=debug,proto=debug,nano=debug,\
-             apiclient=debug",
-        );
+/// Output format emitted to the terminal
+#[derive(Debug, Copy, Clone, PartialEq, Eq, clap::ArgEnum)]
+pub enum OutputFormat {
+    /// Human-readable, single-line logs for each event that occurs, with the current span context
+    /// displayed before the formatted representation of the event.
+    Full,
+    /// excessively pretty, multi-line logs, optimized for human readability.
```

Comment:
This might just be a matter of preference (so we should get @litvand to review too). But I'm not sure `pretty` is _that_ optimised for human readability. I honestly find it hard to immediately find the information I want, I think there's too much bloat.

The information is good to have, probably especially in the JSON format, but the timings for instance take up so much essential space at the start of the line, and are down to a ridiculous accuracy for human consumption. The actual contents of the log that I want to read are hidden way at the end of the line, so when I'm scrolling through thousands of lines of trace log it becomes an unmanageable task

Reply:
Actually the description is more or less taken from the tracing library itself. In tracing, however, the default is `Full`. May we want to swtich to `Full` then?

Reply:
Upon looking at it further, I think the main culprit is the timings. I reckon we could completely get rid of them on non-json logs. Or if we do keep them, maybe we need to reorder it to be at the end of each log

Reply:
Maybe swapping to `Full` will be okay for now and when we fix the JSON output we can look at revisiting the formats.

Reply:
(being done in follow-up PR)

Reply:
I agree the timings are too verbose. `pretty` looks alright though, since it highlights the actual message in a different color.